pkg/cmd/admin/node/node.go

```diff
@@ -0,0 +1,385 @@
+
+Examples:
+
+	# Block the node accepting any new pods
+	$ %[1]s manage-node <mynode> --schedulable=false
+
+	# Move selected pods to a different node
+	$ %[1]s manage-node <mynode> --evacuate --pod-selector="service=myapp"
+
+	# List all pods on nodes
+	$ %[1]s manage-node <mynode1> <mynode2> --list-pods
+
+	# Mark selected nodes as schedulable
+	$ %[1]s manage-node --selector="env=dev" --schedulable=true
+`
+
+func NewCommandManageNode(f *clientcmd.Factory, fullName, commandName string, out io.Writer) *cobra.Command {
+	cfg := NewNodeConfig()
+
+	cmd := &cobra.Command{
+		Use:   commandName,
+		Short: "Manage node operations: schedulable, evacuate, list-pods",
+		Long:  fmt.Sprintf(manageNodeLongDesc, fullName),
+		Run: func(c *cobra.Command, args []string) {
+			err := cfg.RunManageNode(f, out, c, args)
+			kcmdutil.CheckErr(err)
+		},
+	}
+	flags := cmd.Flags()
+
+	flags.BoolVar(&cfg.Schedulable, "schedulable", true, "Control pod schedulability on the node.")
+	flags.BoolVar(&cfg.Evacuate, "evacuate", true, "Migrate all/selected pods on the node.")
+	flags.BoolVar(&cfg.ListPods, "list-pods", true, "List all/selected pods on the node.")
+
+	flags.StringVar(&cfg.PodSelector, "pod-selector", "", "Label selector to filter pods on the node. Optional parameters for --evacuate or --list-pods")
+	flags.BoolVar(&cfg.DryRun, "dry-run", false, "Don't perform evacuate but show pods that will be migrated. Optional param for --evacuate")
+	flags.StringVar(&cfg.Selector, "selector", "", "Label selector to filter nodes. Either pass one/more nodes as arguments or use this node selector")
+
+	return cmd
+}
+
+// RunManageNode contains all the necessary functionality for the openshift cli manage-node command
+func (cfg *NodeConfig) RunManageNode(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args []string) error {
+	errList := []error{}
+	if errList = cfg.ValidateParams(cmd, args); len(errList) != 0 {
+		err := fmt.Errorf("(%v)", errList)
+		return kcmdutil.UsageError(cmd, err.Error())
+	}
+	if err := cfg.Configure(f, out); err != nil {
+		return err
+	}
+
+	cmdNamespace, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+	nameArgs := []string{"nodes"}
+	if len(args) != 0 {
+		nameArgs = append(nameArgs, args...)
+	}
+
+	mapper, typer := f.Object()
+	r := resource.NewBuilder(mapper, typer, f.ClientMapperForCommand()).
+		ContinueOnError().
+		NamespaceParam(cmdNamespace).
+		SelectorParam(cfg.Selector).
+		ResourceTypeOrNameArgs(true, nameArgs...).
+		Flatten().
+		Do()
+	if r.Err() != nil {
+		return r.Err()
+	}
+
+	nodeCount := 0
+	ignoreHeaders := false
+	err = r.Visit(func(info *resource.Info) error {
+		nodeCount++
+		err = nil
+		node, ok := info.Object.(*kapi.Node)
+		if !ok {
+			err = fmt.Errorf("cannot convert input to Node: ", reflect.TypeOf(info.Object))
+			errList = append(errList, err)
+			return nil
+		}
+
+		if cmd.Flag("schedulable").Changed {
```

Comment:
We used "Changed" in a few other places.  @fabianofranz had been working on getting this fixed in upstream pflags as well.  

I'd prefer it pulled out and moved up in the file, but we can leave the construct for now.

----- Original Message -----
> > +		return r.Err()
> > +	}
> > +
> > +	nodeCount := 0
> > +	ignoreHeaders := false
> > +	err = r.Visit(func(info *resource.Info) error {
> > +		nodeCount++
> > +		err = nil
> > +		node, ok := info.Object.(*kapi.Node)
> > +		if !ok {
> > +			err = fmt.Errorf("cannot convert input to Node: ",
> > reflect.TypeOf(info.Object))
> > +			errList = append(errList, err)
> > +			return nil
> > +		}
> > +
> > +		if cmd.Flag("schedulable").Changed {
> 
> @smarterclayton This ties us to `pflags`.  Do we want to go with the way
> we've done `Provided` methods on other commands or is having to explicitly
> set `=true` too annoying?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/2151/files#r30033302